Realm/ObjectServerTests/SwiftFlexibleSyncServerTests.swift

```diff
@@ -941,6 +941,289 @@ class SwiftFlexibleSyncTests: SwiftSyncTestCase {
+            subscriptions.append(QuerySubscription<SwiftHugeSyncObject>())
+        })
+        config.objectTypes = objectTypes
+        var downloadRealm: Realm?
+        let task = Realm.asyncOpen(configuration: config) { result in
+            XCTAssertNotNil(try? result.get())
+            downloadRealm = try! result.get()
+            asyncOpenEx.fulfill()
+        }
+
+        var callCount = 0
+        var progress: SyncSession.Progress?
+
+        task.addProgressNotification { p in
+            DispatchQueue.main.async {
+                if let progress = progress {
+                    if progress.progressEstimate < 1.0 {
+                        XCTAssertGreaterThanOrEqual(p.progressEstimate, progress.progressEstimate)
+                    }
+                }
+                progress = p
+                callCount += 1
+            }
+        }
+
+        waitForExpectations(timeout: 10.0, handler: nil)
+
+        XCTAssertEqual(try XCTUnwrap(downloadRealm).objects(SwiftHugeSyncObject.self).count, 2)
+
+        let p1 = try XCTUnwrap(progress)
+        XCTAssertEqual(p1.progressEstimate, 1.0)
+        XCTAssertTrue(p1.isTransferComplete)
+    }
+
+    func testNonStreamingDownloadNotifier() throws {
+        try populateRealm()
+
+        let realm = try openRealm(wait: false)
+
+        let session = try XCTUnwrap(realm.syncSession)
+        var ex = expectation(description: "first download")
+        var callCount = 0
+        var progress: SyncSession.Progress?
+        let token = session.addProgressNotification(for: .download, mode: .forCurrentlyOutstandingWork) { p in
+            DispatchQueue.main.async {
+                // Verify that progress increases.
+                if let progress = progress {
+                    XCTAssertGreaterThanOrEqual(p.progressEstimate, progress.progressEstimate)
+                }
+                progress = p
+                callCount += 1
+            }
+        }
+        XCTAssertNotNil(token)
+
+        let subscriptions = realm.subscriptions
+        subscriptions.update({
+            subscriptions.append(QuerySubscription<SwiftHugeSyncObject>(name: "huge_objects"))
+        }, onComplete: { err in
+            DispatchQueue.main.async {
+                XCTAssertNil(err)
+                ex.fulfill()
+            }
+        })
+
+        waitForExpectations(timeout: 60.0)
+
+        XCTAssertEqual(realm.objects(SwiftHugeSyncObject.self).count, SwiftSyncTestCase.bigObjectCount)
+
+        XCTAssertGreaterThanOrEqual(callCount, 1)
+        let p1 = try XCTUnwrap(progress)
+        XCTAssertEqual(p1.progressEstimate, 1.0)
+        XCTAssertTrue(p1.isTransferComplete)
+        let initialCallCount = callCount
+        progress = nil
+
+        // Run a second time to upload more data and verify that the callback continues to be called
+        ex = expectation(description: "second download")
+        try populateRealm()
+
+        session.wait(for: .download) { e in
```

Comment:
These should all be able to be just `waitForDownloads(for: realm)` or `waitForUploads(for: realm)`?